enhancements/installer/signal-cluster-deletion.md

```diff
@@ -0,0 +1,121 @@
+---
+title: signal-cluster-deletion
+authors:
+  - "@abutcher"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-03-30
+last-updated: 2020-03-30
+status: provisional
+---
+
+# Signal Cluster Deletion
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Operators can create external resources in clouds and other services that have no opportunity to be cleaned up when a cluster is uninstalled, causing extra work for cluster maintainers to track down and remove these resources. Signalling cluster deletion would allow operators to clean up external resources during uninstallation and would ensure that total cleanup is an automatic process.
+
+## Motivation
+
+### Goals
+
+Signal that cluster deletion has begun through deletion of a new CRD `ClusterAlive` and wait for that object to be deleted during uninstall before continuing with removing cluster resources such as instances, storage and networking.
+
+### Non-Goals
+
+Attaching finalizers to operator resources based on the `ClusterAlive` object which would facilitate removal of operator resources.
```

Comment:
Somewhere in this I'd like to flesh out what we expect to happen here. I believe the proposal from OLM was that operators set finalizers on their CRs, and OLM would set a finalizer on ClusterAlive. When deleted, it would delete every CR for every CRD for every operator, giving them their chance to cleanup resources.

This would still work with regard to @abhinavdahiya 's request for an admin override to block this. Assuming ClusterAlive.Spec.BlockTeardown = true, OLM could respect this flag and refuse to start it's teardown process, we only need to check BlockTeardown in one place.

If however operators start adding their finalizers directly to ClusterAlive, we would need to guarantee that they all also knew to respect this additional BlockTeardown field. The OLM layer of indirection seems required so I'd like to make sure they're still on board with that plan to use CRs+CRDs.

cc @ecordell